Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs

```diff
@@ -876,22 +876,29 @@ private class ExpressionCompletionWalker : PythonWalker {
             DismissCompletionSession();
 
             var session = _services.CompletionBroker.TriggerCompletion(_textView);
-
             if (session == null) {
                 Volatile.Write(ref _activeSession, null);
-            } else if (completeWord && SelectSingleBestCompletion(session)) {
+                return;
+            }
+
+            if (completeWord && SelectSingleBestCompletion(session)) {
                 session.Commit();
-            } else {
-                if (commitByDefault.HasValue) {
-                    foreach (var s in session.CompletionSets.OfType<FuzzyCompletionSet>()) {
-                        s.CommitByDefault = commitByDefault.GetValueOrDefault();
-                    }
+                return;
+            }
+
+            if (completeWord) {
+                session.SetCompleteWordMode();
```

Comment:
Don't we need to set this before the first round of completions are calculated? (When we chatted I suggested splitting the TriggerCompletion call into separate "create completion" and "start completion" calls, so we can set these properties in the middle.)

Reply:
Unfortunately, session is available only after `broker.TriggerCompletion()` returns. So I use both property and the `IntellisenseController.ForceCompletions` in the extension.

So we want to split Trigger into `ICompletionBroker.CreateCompletion` and the start right?

Reply:
Yeah. That's all that Trigger actually does, as well as handling failure to start safely. It's only 3-4 LoC.

Reply:
I've done that, pending running tests